NitroxServer/GameLogic/Bases/BaseManager.cs

```diff
@@ -68,8 +108,16 @@ public void BasePieceConstructionCompleted(NitroxId id, NitroxId baseId)
             {
                 if (partiallyConstructedPiecesById.TryGetValue(id, out basePiece))
                 {
-                    basePiece.ConstructionAmount = 1.0f;
-                    basePiece.ConstructionCompleted = true;
+                    if (basePiece.ConstructionAmount > 0.4f)
+                    {
+                        basePiece.ConstructionAmount = 1.0f;
+                        basePiece.ConstructionCompleted = true;
+                    }
+                    else
+                    {
+                        basePiece.ConstructionAmount = 1.0f;
+                        basePiece.ConstructionCompleted = true;  
```

Comment:
Let's not split up the code into an if/else with same content in it, unfinish maybe ?